packages/typescript-client/test/integration.test.ts

```diff
@@ -163,6 +163,202 @@ describe(`HTTP Sync`, () => {
+      DROP TYPE IF EXISTS complex;
+      DROP DOMAIN IF EXISTS posint;
+      CREATE TYPE mood AS ENUM ('sad', 'ok', 'happy');
+      CREATE TYPE complex AS (r double precision, i double precision);
+      CREATE DOMAIN posint AS integer CHECK (VALUE > 0);
+      CREATE TABLE ${table} (
+        txt VARCHAR,
+        i2 INT2 PRIMARY KEY,
+        i4 INT4,
+        i8 INT8,
+        f8 FLOAT8,
+        b  BOOLEAN,
+        json JSON,
+        jsonb JSONB,
+        ints INT8[],
+        ints2 INT8[][],
+        int4s INT4[],
+        bools BOOLEAN[],
+        moods mood[],
+        moods2 mood[][],
+        complexes complex[],
+        posints posint[],
+        jsons JSONB[],
+        txts TEXT[],
+        value JSON
+      )
+    `)
+
+    await dbClient.query(
+      `
+      INSERT INTO ${table} (txt, i2, i4, i8, f8, b, json, jsonb, ints, ints2, int4s, bools, moods, moods2, complexes, posints, jsons, txts, value)
+      VALUES (
+        'test',
+        1,
+        2,
+        3,
+        4.5,
+        TRUE,
+        '{"foo": "bar"}',
+        '{"foo": "bar"}',
+        '{1,2,3}',
+        $1,
+        $2,
+        $3,
+        $4,
+        $5,
+        $6,
+        $7,
+        $8,
+        $9,
+        $10
+      )
+    `,
+      [
+        [
+          [1, 2, 3],
+          [4, 5, 6],
+        ],
+        [1, 2, 3],
+        [true, false, true],
+        [`sad`, `ok`, `happy`],
+        [
+          [`sad`, `ok`],
+          [`ok`, `happy`],
+        ],
+        [`(1.1, 2.2)`, `(3.3, 4.4)`],
+        [5, 9, 2],
+        [{ foo: `bar` }, { bar: `baz` }],
+        [`foo`, `bar`, `baz`],
+        { a: 5, b: [{ c: `foo` }] },
+      ]
+    )
+
+    // Now fetch the data from the HTTP endpoint
+    const issueStream = new ShapeStream({
+      shape: { table },
+      baseUrl: `${BASE_URL}`,
+      signal: aborter.signal,
+    })
+    const client = new Shape(issueStream)
+    const data = await client.value
+
+    expect([...data.values()]).toMatchObject([
+      {
+        txt: `test`,
+        i2: 1,
+        i4: 2,
+        i8: BigInt(3),
+        f8: 4.5,
+        b: true,
+        json: { foo: `bar` },
+        jsonb: { foo: `bar` },
+        ints: [BigInt(1), BigInt(2), BigInt(3)],
+        ints2: [
+          [BigInt(1), BigInt(2), BigInt(3)],
+          [BigInt(4), BigInt(5), BigInt(6)],
+        ],
+        int4s: [1, 2, 3],
+        bools: [true, false, true],
+        moods: [`sad`, `ok`, `happy`],
+        moods2: [
+          [`sad`, `ok`],
+          [`ok`, `happy`],
+        ],
+        // It does not parse composite types and domain types
+        complexes: [`(1.1,2.2)`, `(3.3,4.4)`],
+        posints: [`5`, `9`, `2`],
+        jsons: [{ foo: `bar` }, { bar: `baz` }],
+        txts: [`foo`, `bar`, `baz`],
+        value: { a: 5, b: [{ c: `foo` }] },
+      },
+    ])
+
+    // Now update the data
+    await dbClient.query(
+      `
+      UPDATE ${table}
+      SET
+        txt = 'changed',        
```

Comment:
looking at this custom serializer/parser — tests for `Infinity` & `NaN` would be good as they're supported in both postgres and js https://github.com/Rich-Harris/devalue

Reply:
Good idea, will add!